src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -218,6 +218,27 @@ import scala.runtime.Statics
   *  filtered.isEmpty // prints "getting empty LazyList"
   *  }}}
   *
+  *  ----
+  *
+  *  You may sometimes encounter an exception like the following:
+  *
+  *  {{{
+  *  java.lang.RuntimeException: "LazyList evaluation depends on its own result (self-reference); see docs for more info
+  *  }}}
+  *
+  *  This exception occurs when a `LazyList` is attempting to derive its next element
+  *  from itself, and is attempting to read the element currently being evaluated. A
+  *  trivial example of such might be
+  *
+  *  {{{
+  *  lazy val a: LazyList[Int] = 1 #:: 2 #:: a.filter(_ > 2)
+  *  }}}
+  *
+  *  When attempting to evaluate the third element of `a`, it will skip the first two
+  *  elements and read the third, but that element is already being evaluated. This is
+  *  often caused by a subtle logic error; in this case, using `>=` in the `filter`
+  *  would fix the error.
```

Comment:
That is very subtle. I will have to fully context-switch to understand it. At the moment, I see from the previous exception message that it has something to do with "no more elements". To borrow a phrase, it's amazing that it works at all.

Reply:
Better provides a general guide section?

Reply:
do you think I should keep "no more elements" in the message?

Reply:
tbh, I'm not happy with my docs either, but they seem better than just the previous message hopefully? and can be incrementally improved by others as well